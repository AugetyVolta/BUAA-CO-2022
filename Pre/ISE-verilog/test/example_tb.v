`timescale  1ns / 1ps
`include "example.v"
module tb_chardet;   

// chardet Parameters
parameter PERIOD  = 10;


// chardet Inputs
reg   a                                    = 0 ;
reg   b                                    = 0 ;
reg   c                                    = 0 ;

// chardet Outputs


chardet  u_chardet (
    .a                       ( a   ),
    .b                       ( b   ),
    .c                       ( c   )
);

initial
begin
    $dumpfile("example.vcd");
    $dumpvars;
    #10
    a=1;
    b=0;
    c=1;
    #10
    a=0;
    b=1;
    c=0;
    #10
    a=1;
    b=1;
    c=0;
    $finish;
end

endmodule