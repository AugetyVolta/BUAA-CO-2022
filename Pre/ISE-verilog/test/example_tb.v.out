#! c:/iverilog-x86/bin/vvp
:ivl_version "10.0 (stable)" "(v10_0)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 12;
:vpi_module "system";
:vpi_module "vhdl_sys";
:vpi_module "v2005_math";
:vpi_module "va_math";
S_00939890 .scope module, "tb_chardet" "tb_chardet" 2 3;
 .timescale -9 -12;
P_00934a20 .param/l "PERIOD" 0 2 6, +C4<00000000000000000000000000001010>;
v009338d8_0 .var "a", 0 0;
v00939ad8_0 .var "b", 0 0;
v00939b30_0 .var "c", 0 0;
S_00933758 .scope module, "u_chardet" "chardet" 2 17, 3 1 0, S_00939890;
 .timescale -9 -12;
    .port_info 0 /INPUT 1 "a"
    .port_info 1 /INPUT 1 "b"
    .port_info 2 /INPUT 1 "c"
v00934018_0 .net "a", 0 0, v009338d8_0;  1 drivers
v00933828_0 .net "b", 0 0, v00939ad8_0;  1 drivers
v00933880_0 .net "c", 0 0, v00939b30_0;  1 drivers
    .scope S_00939890;
T_0 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v009338d8_0, 0, 1;
    %end;
    .thread T_0;
    .scope S_00939890;
T_1 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00939ad8_0, 0, 1;
    %end;
    .thread T_1;
    .scope S_00939890;
T_2 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00939b30_0, 0, 1;
    %end;
    .thread T_2;
    .scope S_00939890;
T_3 ;
    %vpi_call 2 25 "$dumpfile", "example.vcd" {0 0 0};
    %vpi_call 2 26 "$dumpvars" {0 0 0};
    %delay 10000, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v009338d8_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00939ad8_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v00939b30_0, 0, 1;
    %delay 10000, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v009338d8_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v00939ad8_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00939b30_0, 0, 1;
    %delay 10000, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v009338d8_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v00939ad8_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00939b30_0, 0, 1;
    %vpi_call 2 39 "$finish" {0 0 0};
    %end;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "example_tb.v";
    "./example.v";
